ttnn/cpp/ttnn/operations/moreh/moreh_getitem/device/moreh_getitem_device_operation.cpp

```diff
@@ -23,14 +23,14 @@ void MorehGetItemOperation::validate_inputs(
         dtype == DataType::INT32 || dtype == DataType::BFLOAT16, "Input tensor must be of type INT32 or BFLOAT16!");
 
     // validate index tensors
-    uint32_t index_size = index_tensors[0].get_shape()[-1];
+    const uint32_t& index_size = index_tensors[0].get_logical_shape()[-1];
```

Comment:
no need for const& here. you are getting a specific value our of the shape
```suggestion
    const uint32_t index_size = index_tensors[0].get_logical_shape()[-1];
```